### ISSUE TITLE
Set up  a static site generator for the community site

### ISSUE THREAD

**Issue by samccann** [ansible-community/community-team#166]

**Comment by leogallego** [comment]
Idea: Maybe use a static generator based on Python with Jinja (and possible Markdown) templating, to keep in line with Ansible coders/users skills if required? 

Quick draft list to check:
* https://hyde.github.io/
* https://www.getlektor.com/
* https://getpelican.com/

**Comment by maxamillion** [comment]
+1 for [pelican|https://getpelican.com/].

It has a massive global community and is in use by noteworthy open source projects such as [the linux kernel|https://www.kernel.org/pelican.html]. I think it would be nice to standardize on pelican for all Ansible Community website needs.

**Comment by shanemcd** [comment]
I think we should consider Sphinx. It's what we already use for most of our upstream projects, and have a good number of people who are familiar with it.

**Comment by maxamillion** [comment]
> I think we should consider Sphinx. It's what we already use for most of our upstream projects, and have a good number of people who are familiar with it.

@shanemcd we use it for generating docs everywhere, but I'm not sure it's a good tool to build a general purpose website out of. Do you know of a good example of sphinx being used for that purpose?

**Comment by samccann** [comment]
From discussion:
- needs to be responsive design
- support dynamic elements such as a blog
-

**Comment by oraNod** [comment]
@leogallego @maxamillion @shanemcd @samccann 

As a next step let's identify things we need/want from a SSG. Please take a look at this list of required / nice to haves and add your 2c: https://hackmd.io/n5nBt-nlTvaBEDCUXrnZHQ?edit

We can then evaluate Sphinx, Lektor, Pelican, et al against our list. Thank you!

**Comment by oraNod** [comment]
We've also taken this to the Ansible community to make sure their voices are represented in the SSG decision: https://github.com/ansible-community/community-topics/issues/210

**Comment by oraNod** [comment]
@maxamillion Hey, we've got a checklist of requirements and nice to haves for the static site generator in the community topic: https://github.com/ansible-community/community-topics/issues/210

Could you please take a look and let me know if you have any thoughts / nits about the checklist itself? I'd also appreciate if you could add a comment to that topic to assess Pelican against the checklist. Thank you!!

**Comment by oraNod** [comment]
@leogallego Do you think you could add a comment that assesses Hyde against the SSG checklist at https://github.com/ansible-community/community-topics/issues/210 ?

**Comment by maxamillion** [comment]
@oraNod I think the CMS-style editing being a requirement is rather prohibitive for majority of static site generators, including almost all the most popular ones. Other than that, I think the list is great!

**Comment by leogallego** [comment]
@oraNod I feel Hyde doesn't have an active community or is actively maintained. It seems their last commit was in 2016 and their latest release was 0.8.9 in 2015. On that alone I would exclude it. I don't think it even works with Python 3.x

@maxamillion the CMS/web-ui was a nice-to-have originally, but we removed it in the last revision of the checklis. I don't see it in the current checklist in the issue, you might have had an old version open?

**Comment by maxamillion** [comment]
> @maxamillion the CMS/web-ui was a nice-to-have originally, but we removed it in the last revision of the checklis. I don't see it in the current checklist in the issue, you might have had an old version open? 

yeah, sorry ... I should have refreshed my browser before posting because it seems my comment is not relevant anymore.

**Comment by oraNod** [comment]
@shanemcd Hey, I added a comment for Sphinx to the topic. Could you please take a look and add any thoughts/nits? https://github.com/ansible-community/community-topics/issues/210#issuecomment-1478107326
